### PR TITLE
Feature/update code-style and adapt code

### PR DESCRIPTION
**Pull request by SuperITMan**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[X] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

`stylelint` and `codelyzer` are not up to date.

## What is the new behavior?

`stylelint`, `codelyzer` and `nationalbankbelgium/code-style` are up to date.

Code has been fixed and reformated when necessary.

## Does this PR introduce a breaking change?
```
[X] Yes
[ ] No
```

## Other information

### CODE REVIEW

**Comment by nicanac** [comment]
So far so good, could you adapt the commit message ?
thanks

**Comment by SuperITMan** [comment]
@nicanac commit messages adapted

**Comment by nicanac** [comment]
> @nicanac commit messages adapted

Perfect

File: packages/stark-ui/assets/styles/components/_button-theme.scss [hunk 1]
```diff
@@ -23,7 +23,7 @@
       background-color: rgba($color: $contrast, $alpha: 0.1);
     }
   }
-  .mat-icon-button {
+  &.mat-icon-button {
```
Comment (christophercr) [inline packages/stark-ui/assets/styles/components/_button-theme.scss]: Are we sure of this change? `.xxx` is not the same as `&.xxx`

File: packages/stark-ui/src/modules/action-bar/components/_action-bar.component.scss [hunk 2]
```diff
@@ -11,6 +11,13 @@
   overflow: hidden;
   transform-origin: bottom;
 
+  &.extended {
+    .stark-action-bar-action {
```
Comment (christophercr) [inline packages/stark-ui/src/modules/action-bar/components/_action-bar.component.scss]: What is this change about? There is just added code... no refactored code 🤔

File: packages/stark-ui/src/modules/action-bar/components/action-bar.component.html [hunk 3]
```diff
@@ -5,35 +5,29 @@
...
-				>
-					<mat-icon
-						[svgIcon]="action.iconSwitchFunction ? action.iconActivated : action.icon"
-						class="stark-small-icon"
-					></mat-icon>
-				</button>
-				<label
+		<ng-container *ngFor="let action of actionBarConfig.actions; trackBy: trackAction">
+			<button
+				[id]="actionBarId + '-' + action.id"
```
Comment (christophercr) [inline packages/stark-ui/src/modules/action-bar/components/action-bar.component.html]: Why this change? Now the one with `id` is the `<button>` and before it was the `<div>`... is this something intended? Is this tested?

File: packages/stark-ui/src/modules/action-bar/components/action-bar.component.html [hunk 4]
```diff
@@ -58,20 +52,20 @@
...
-				<ng-container *ngIf="action.isVisible !== false">
-					<div [ngClass]="action.className">
-						<mat-icon [svgIcon]="action.icon" class="stark-small-icon"></mat-icon>
-						<span translate>{{ action.label }}</span>
-					</div>
-				</ng-container>
-			</div>
+			<ng-container *ngFor="let action of alternativeActions; trackBy: trackAction">
+				<button
+					*ngIf="action.isVisible !== false"
```
Comment (christophercr) [inline packages/stark-ui/src/modules/action-bar/components/action-bar.component.html]: Same here: is this intended? is it tested?

File: packages/stark-ui/src/modules/table/components/_table.component.scss [hunk 5]
```diff
@@ -130,6 +120,16 @@
           }
         }
       }
+
+      &.filtering {
+	&.mat-header-cell {
```
Comment (christophercr) [inline packages/stark-ui/src/modules/table/components/_table.component.scss]: It seems this line is not formatted

File: showcase/.stylelintrc [hunk 6]
```diff
@@ -1,5 +1,5 @@
 {
-  	"extends": ["@nationalbankbelgium/code-style/stylelint/9.10.x", "stylelint-config-prettier"],
+	"extends": ["stylelint-config-prettier", "@nationalbankbelgium/code-style/stylelint/13.0.x"],
```
Comment (christophercr) [inline showcase/.stylelintrc]: `stylelint-config-prettier` should be at the end of the array

File: starter/.stylelintrc [hunk 7]
```diff
@@ -1,3 +1,3 @@
 {
-  "extends": ["@nationalbankbelgium/code-style/stylelint/9.10.x", "stylelint-config-prettier"]
+  "extends": ["stylelint-config-prettier", "@nationalbankbelgium/code-style/stylelint/13.0.x"]
```
Comment (christophercr) [inline starter/.stylelintrc]: Same remark: `stylelint-config-prettier` should be at the end of this array

File: showcase/src/app/demo-rbac/services/demo-authorization.service.ts [hunk 8]
```diff
@@ -2,7 +2,7 @@ import { StarkRBACAuthorizationServiceImpl } from "@nationalbankbelgium/stark-rb
 import { StarkUser } from "@nationalbankbelgium/stark-core";
 import { Injectable } from "@angular/core";
 
-@Injectable()
+@Injectable({ providedIn: "root" })
```
Comment (christophercr) [inline showcase/src/app/demo-rbac/services/demo-authorization.service.ts]: Weird that this service was not included in the  `providers` of the module and yet it was working!! 😲

File: packages/stark-ui/src/modules/message-pane/components/_message-pane.component.scss [hunk 9]
```diff
@@ -197,9 +179,23 @@
       }
     }
   }
+}
 
-  & mat-icon {
-    transition: all 0.4s;
+.stark-app-header .stark-message-pane {
```
Comment (christophercr) [inline packages/stark-ui/src/modules/message-pane/components/_message-pane.component.scss]: Just double checking... are the changes in this file not breaking anything on the styling of the component? 
Cause I cannot see clearly how this has changed... the diff view is not helping :p

File: docs/stark-build/NG_CLI_BUILD_CUSTOMIZATIONS.md [hunk 10]
```diff
@@ -292,11 +292,7 @@ the following lines in your `<head>` section:
 
 ```html
 <head>
-	...
-	<% if (webpackConfig.htmlElements.headTags) { %>
-	<%= webpackConfig.htmlElements.headTags %>
-	<% } %>
-	...
+	... <% if (webpackConfig.htmlElements.headTags) { %> <%= webpackConfig.htmlElements.headTags %> <% } %> ...
```
[the remark this answers is not in the document]
Comment (SuperITMan) [reply docs/stark-build/NG_CLI_BUILD_CUSTOMIZATIONS.md]: Good idea! 😊

File: packages/stark-ui/assets/styles/components/_button-theme.scss (hunk 1 above)
Comment (SuperITMan) [reply packages/stark-ui/assets/styles/components/_button-theme.scss]: Actually, the rule was incorrect before. The `mat-icon-class` is added on the same element than the `mat-button` class. It's a fix 😊

File: .stylelintrc [hunk 11]
```diff
@@ -1,3 +1,3 @@
 {
-	"extends": ["@nationalbankbelgium/code-style/stylelint/9.10.x", "stylelint-config-prettier"]
+  "extends": ["stylelint-config-prettier", "@nationalbankbelgium/code-style/stylelint/13.0.x"]
```
[the remark this answers is not in the document]
Comment (SuperITMan) [reply .stylelintrc]: Yeah! My bad. I fix that now 😊

File: packages/stark-ui/src/modules/table/components/_table.component.scss (hunk 5 above)
Comment (SuperITMan) [reply packages/stark-ui/src/modules/table/components/_table.component.scss]: Indeed 👍

File: showcase/src/app/demo-rbac/services/demo-authorization.service.ts (hunk 8 above)
Comment (SuperITMan) [reply showcase/src/app/demo-rbac/services/demo-authorization.service.ts]: You're right ^^
Sometimes, angular is magic :sparkles:

File: packages/stark-ui/src/modules/action-bar/components/action-bar.component.html (hunk 3 above)
Comment (SuperITMan) [reply packages/stark-ui/src/modules/action-bar/components/action-bar.component.html]: Actually, the previous implementation was wrong for 2 reasons. The `(click)` should be on the `<button>` element and not on the `<div>`. Also, when an action is **not** visible, it results like this in the previous implementation (element still exists and the click event also):
image

In the new one, the element is not generated at all:
```html
<!--bindings={
  "ng-reflect-ng-if": "false"
}-->
```

I tested manually the layout and the usability and everything seems ok :blush:

File: packages/stark-ui/src/modules/action-bar/components/action-bar.component.html (hunk 4 above)
Comment (SuperITMan) [reply packages/stark-ui/src/modules/action-bar/components/action-bar.component.html]: Same here: https://github.com/NationalBankBelgium/stark/pull/1648#discussion_r381937580

File: packages/stark-ui/src/modules/action-bar/components/_action-bar.component.scss (hunk 2 above)
Comment (SuperITMan) [reply packages/stark-ui/src/modules/action-bar/components/_action-bar.component.scss]: Due to the change applied in the template (`<label>` replaced by a `<span>` and `<div>` removed), it's needed to apply the `width: auto` when extended.
The `text-transform: none` should be applied in `.stark-action-bar-action.stark-action-bar-full .action-label` actually. I change this 😊

File: packages/stark-ui/src/modules/message-pane/components/_message-pane.component.scss (hunk 9 above)
Comment (SuperITMan) [reply packages/stark-ui/src/modules/message-pane/components/_message-pane.component.scss]: Actually, those changes are due to the added stylelint rule `no-descending-specificity: true`
I simply reorganized well the rules 😊 
According to my tests, it does not change anything

**Comment by SuperITMan** [comment]
@christophercr I updated my PR based on your comments. Could you please have a look on it ? 😊

**Comment by christophercr** [comment]
@SuperITMan the commit message of the second says is `stylelint dependency to 13.0.O` but it should be `stylelint dependency to 13.0.0`

**Comment by SuperITMan** [comment]
@christophercr good catch for the `O` instead of `0` 👍 

@nicanac 
About the commit for codelyzer, in Stark, the update was breaking the build system.
For instance, had the following comment in some "*spec.ts" files `tslint:disable:no-life-cycle-call`. After the codelyzer upgrade, that we do in `stark-build`, it has to be changed to `tslint:disable:no-lifecycle-call`.
That's why I added the list of `codelyzer` changes. What do you think ? Should we keep it or not ?

**Comment by SuperITMan** [comment]
@christophercr I updated the message of my first commit. What do you think of the BREAKING CHANGE message ? 😊 

BREAKING CHANGES: Due to the update of the `codelyzer` dependency, multiple rules have been renamed and new rules been added, as documented on [codelyzer/CHANGELOG.md](https://github.com/mgechev/codelyzer/blob/master/CHANGELOG.md). We strongly recommend to upgrade your `tslint` configuration to use the new rules. The [@nationalbankbelgium/code-style@1.1.1](https://github.com/NationalBankBelgium/code-style/releases/tag/1.1.1) library contains the NBB recommended configuration for `codelyzer@5.2.x`.

**Comment by nicanac** [comment]
> @christophercr good catch for the `O` instead of `0` 👍
> 
> @nicanac
> About the commit for codelyzer, in Stark, the update was breaking the build system.
> For instance, had the following comment in some "*spec.ts" files `tslint:disable:no-life-cycle-call`. After the codelyzer upgrade, that we do in `stark-build`, it has to be changed to `tslint:disable:no-lifecycle-call`.
> That's why I added the list of `codelyzer` changes. What do you think ? Should we keep it or not ?

Yes, I think we must keep it.

File: showcase/src/app/demo-rbac/services/demo-authorization.service.ts (hunk 8 above)
Comment (nicanac) [reply showcase/src/app/demo-rbac/services/demo-authorization.service.ts]: Magic or evil ? :-)

**Comment by christophercr** [comment]
> About the commit for codelyzer, in Stark, the update was breaking the build system.
For instance, had the following comment in some "*spec.ts" files tslint:disable:no-life-cycle-call. After the codelyzer upgrade, that we do in stark-build, it has to be changed to tslint:disable:no-lifecycle-call.

Mmm I don't remember exactly why we have `codelyzer` as dependency in `stark-build`... technically we don't need it there because the build will just pick the TSLint config of the project (which might use codelyzer or not). Or am I missing something? It's been a while since we worked on the `stark-build` :p

So to me, this is not a breaking change for any project because we just changed the internal TSLint configs used in Stark itself... that was one of the reasons to extract the config to `code-style` right? so the projects depend on that package instead of Stark for the linting rules...

And
